### DISCUSSION TITLE
Wait for plugin initialization

### DISCUSSION THREAD

**Discussion by tntsoft** [nuxt/nuxt#17579] [Ideas]
I'd like Nuxt to be able to wait for a plugin to be fully initialised before proceeding - optionally, of course.

Example:
```
export default defineNuxtPlugin(() => {
    return {
        provide: {
            App: new App()
        }
    }
})
```

This App class has a constructor, and that constructor does essential work. I'd like to be able to set a variable/something in this plugin declaration file that will block everything until this App's constructor has fully ran its course - particularly want to block Middleware.

I'm syncing user data through this plugin and would like to wait for it to fully finish before moving on.

**Comment by TheAlexLichter** [comment]
Hey 👋 

You could make the plugin function `async` and `await` a function call of the class setup. Something like:

```js
export default defineNuxtPlugin(async () => {
    const app = new App()
    await app.syncUsers()
    return {
        provide: {
            App: app
        }
    }
})
```

**Reply by tntsoft** [reply]
Great idea - and it works! Thank you! 🥳
